go/consensus/tendermint/tendermint.go

```diff
@@ -587,68 +626,70 @@ func (t *tendermintService) Scheduler() schedulerAPI.Backend {
 func (t *tendermintService) initialize() error {
 	t.Lock()
 	defer t.Unlock()
-	if !t.isInitialized {
-		t.Logger.Debug("Initializing tendermint local node.")
-		if err := t.lazyInit(); err != nil {
-			return err
-		}
 
-		if err := t.initEpochtime(); err != nil {
-			return err
-		}
-		t.mux.SetEpochtime(t.epochtime)
+	if t.isInitialized {
+		return nil
+	}
 
-		// Initialize the rest of backends.
-		var err error
-		if t.beacon, err = tmbeacon.New(t.ctx, t); err != nil {
-			t.Logger.Error("initialize: failed to initialize beacon backend",
-				"err", err,
-			)
-			return err
-		}
+	if err := t.lazyInit(); err != nil {
+		return err
+	}
 
-		if t.keymanager, err = tmkeymanager.New(t.ctx, t); err != nil {
-			t.Logger.Error("initialize: failed to initialize keymanager backend",
-				"err", err,
-			)
-			return err
-		}
+	if err := t.initEpochtime(); err != nil {
+		return err
+	}
+	if err := t.mux.SetEpochtime(t.epochtime); err != nil {
+		return err
+	}
 
-		if t.registry, err = tmregistry.New(t.ctx, t); err != nil {
-			t.Logger.Error("initialize: failed to initialize registry backend",
-				"err", err,
-			)
-			return err
-		}
-		t.registry = registry.NewMetricsWrapper(t.ctx, t.registry)
-		t.svcMgr.RegisterCleanupOnly(t.registry, "registry backend")
+	// Initialize the rest of backends.
+	var err error
+	if t.beacon, err = tmbeacon.New(t.ctx, t); err != nil {
+		t.Logger.Error("initialize: failed to initialize beacon backend",
+			"err", err,
+		)
+		return err
+	}
 
-		if t.staking, err = tmstaking.New(t.ctx, t); err != nil {
-			t.Logger.Error("staking: failed to initialize staking backend",
-				"err", err,
-			)
-			return err
-		}
-		t.svcMgr.RegisterCleanupOnly(t.staking, "staking backend")
+	if t.keymanager, err = tmkeymanager.New(t.ctx, t); err != nil {
+		t.Logger.Error("initialize: failed to initialize keymanager backend",
+			"err", err,
+		)
+		return err
+	}
 
-		if t.scheduler, err = tmscheduler.New(t.ctx, t); err != nil {
-			t.Logger.Error("scheduler: failed to initialize scheduler backend",
-				"err", err,
-			)
-			return err
-		}
-		t.svcMgr.RegisterCleanupOnly(t.scheduler, "scheduler backend")
+	if t.registry, err = tmregistry.New(t.ctx, t); err != nil {
```

Comment:
No more registry metrics?

Reply:
The only metrics were based on actual registry method invocations (and the registry backend now no longer has these methods as all transactions go through `SubmitTx` on the consensus backend). We should probably move these metrics to the consensus part as otherwise you are only counting local invocations.

Reply:
it did actually also periodically check `updatePeriodicMetrics` (which posted number of registered nodes and entities). But i do agree that that actually didn't even belong in the metrics wrapper and should be done somewhere else :) (would be nice to keep those metrics though)

Reply:
Oh I didn't see that, I'll add it back :-)

Reply:
Added back (just the periodic updater), PTAL.

Reply:
looks good :+1: